recipes/openjdk/all/conanfile.py

```diff
@@ -12,42 +13,53 @@ class OpenJDK(ConanFile):
 
     def build(self):
-        tools.get(**self.conan_data["sources"][self.version][str(self.settings.os)],
-                  destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version][str(self.settings.os)],
+                  destination=self.source_folder, strip_root=True)
 
     def package(self):
         if self.settings.os == "Macos":
-            _source_subfolder = os.path.join(self._source_subfolder, "jdk-{}.jdk".format(self.version), "Contents", "Home")
+            source_folder = os.path.join(self.source_folder, f"jdk-{self.version}.jdk", "Contents", "Home")
         else:
-            _source_subfolder = self._source_subfolder
-        self.copy(pattern="*", dst="bin", src=os.path.join(_source_subfolder, "bin"),
-                  excludes=("msvcp140.dll", "vcruntime140.dll", "vcruntime140_1.dll"))
-        self.copy(pattern="*", dst="include", src=os.path.join(_source_subfolder, "include"))
-        self.copy(pattern="*", dst="lib", src=os.path.join(_source_subfolder, "lib"))
-        self.copy(pattern="*", dst=os.path.join("lib", "jmods"), src=os.path.join(_source_subfolder, "jmods"))
-        self.copy(pattern="*", dst="licenses", src=os.path.join(_source_subfolder, "legal"))
+            source_folder = self.source_folder
+        symlinks.remove_broken_symlinks(self, source_folder)
+        copy(self, pattern="*",
+                src=os.path.join(source_folder, "bin"),
+                dst=os.path.join(self.package_folder, "bin"),
+                excludes=("msvcp140.dll", "vcruntime140.dll", "vcruntime140_1.dll"))
+        copy(self, pattern="*",
+                src=os.path.join(source_folder, "include"),
+                dst=os.path.join(self.package_folder, "include"))
+        copy(self, pattern="*",
+                src=os.path.join(source_folder, "lib"),
+                dst=os.path.join(self.package_folder, "lib"))
+        copy(self, pattern="*",
+                src=os.path.join(source_folder, "jmods"),
+                dst=os.path.join(self.package_folder, "lib", "jmods"))
+        copy(self, pattern="*",
+                src=os.path.join(source_folder, "legal"),
+                dst=os.path.join(self.package_folder, "licenses"))
         # conf folder is required for security settings, to avoid
         # java.lang.SecurityException: Can't read cryptographic policy directory: unlimited
         # https://github.com/conan-io/conan-center-index/pull/4491#issuecomment-774555069
-        self.copy(pattern="*", dst="conf", src=os.path.join(_source_subfolder, "conf"))
+        copy(self, pattern="*",
+                src=os.path.join(source_folder, "conf"),
+                dst=os.path.join(self.package_folder, "conf"))
 
     def package_info(self):
-        self.output.info("Creating JAVA_HOME environment variable with : {0}".format(self.package_folder))
-        self.env_info.JAVA_HOME = self.package_folder
```

Comment:
Keep env_info.JAVA_HOME, https://conan.io/center/openapi-generator is still not ported and will not work without it.

Reply:
Done here: 41fcb72a7

Reply:
I thought I wrote the [openapi-generator recipe](https://github.com/conan-io/conan-center-index/pull/13691) in a conan v2 compatible way.